fuel/datasets/binarized_mnist.py

```diff
@@ -1,19 +1,14 @@
 # -*- coding: utf-8 -*-
 import logging
 import os
-from collections import OrderedDict
-
-import numpy
 
 from fuel import config
-from fuel.datasets import IndexableDataset
-from fuel.utils import do_not_pickle_attributes
+from fuel.datasets import H5PYDataset
 
 logger = logging.getLogger(__name__)
 
 
-@do_not_pickle_attributes('indexables')
-class BinarizedMNIST(IndexableDataset):
+class BinarizedMNIST(H5PYDataset):
```

Comment:
I was mostly thinking in terms of handling the multiple file opening conundrum, but if we've got that. What I'm mostly against is implementation details leaking into.

@bartvm is right right in that delegation leaves the door open to multiple backends but I suppose that Python's a dynamic language and we can change the inheritance tree later without many adverse effects, so the play in that possible future is basically to change datasets to inherit from something other than H5PYDataset and then make _that_ class do the delegation.